static/app/components/searchSyntax/parser.tsx

```diff
@@ -600,9 +629,102 @@ export class TokenConverter {
       return this.checkInvalidInFilter(value as InFilter['value']);
     }
 
+    const isAggregateFilter = [
+      FilterType.AggregateDate,
+      FilterType.AggregateDuration,
+      FilterType.AggregateNumeric,
+      FilterType.AggregatePercentage,
+      FilterType.AggregateRelativeDate,
+    ].includes(filter);
+
+    if (isAggregateFilter) {
+      return this.checkInvalidAggregateFilter(
+        filter as AggregateFilter,
+        key as FilterMap[AggregateFilter]['key'],
+        value as FilterMap[AggregateFilter]['value']
+      );
+    }
+
     return null;
   };
 
+  checkInvalidAggregateFilter = (
+    filterType: AggregateFilter,
+    key: FilterMap[AggregateFilter]['key'],
+    value: FilterMap[AggregateFilter]['value']
+  ) => {
+    const {isNumeric, isDuration, isDate, isPercentage} = this.keyValidation;
+
+    const keyName = key.name.text;
+
+    const checkAggregateKey = (check: (s: string) => boolean) => {
+      const definition = getFieldDefinition(keyName);
+
+      if (check(keyName)) {
+        if (key.args === null) {
+          // Aggregate keys without args will rely on default values
+          return null;
+        }
+
+        // Get valid column types from the aggregate key
+        const aggregation = AGGREGATIONS[keyName];
```

Comment:
Told you this in person but just wanted to note that this is an `any`

The types here are a big gnarly though so I'm okay with that for now